src/main/perl/FileEditor.pm

```diff
@@ -58,13 +66,45 @@ sub _is_valid_source
     return -f $fn;
 }
 
+sub _is_reference_newer
+{
+    my ($self) = @_;
+    my $is_newer = 0;   # Assume false
+    if (  exists(*$self->{options}->{source}) && $self->_is_valid_source(*$self->{options}->{source}) ) {
```

Comment:
ok, this is potentially dangerous, as `FileReader` inherits from Editor, but `valid_source` for reader can be a pipe, and the stat code below will not like that. so this PR needs modifcations for FileReader (unless you want editor to support pipes as `source`, which could make sense).